File: FindKthlargestElement.java
// Time Complexity : O(N) - As we traverse through all the elements of the array
// Space Complexity : O(N) - As we use a  priority queue data structure
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> queue = new PriorityQueue<>((a,b) -> b - a);
        for(int i = 0;i <nums.length;i++){
            queue.add(nums[i]);
        }
        for(int j = 0;j < k - 1;j++){
            int m = queue.poll();
        }
        return queue.peek();
    }
}
// Your code here along with comments explaining your approach

File: find_kth_largest_element.py
# Time Complexity : O(N log K) where N is number of elements given
# Space Complexity : O(1)
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : No

import heapq as hq

class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        ktop = []
        ktop.append(nums[0])
        hq.heapify(ktop)
        
        for i in range(1, len(nums)):
            if len(ktop) == k:
                if ktop[0] < nums[i]:
                    hq.heappop(ktop)
                    hq.heappush(ktop, nums[i])
            else:
                hq.heappush(ktop, nums[i])
                
        return ktop[0]